src/eda_server/project.py

```diff
@@ -3,114 +3,305 @@
+    raw_content: str
+    content: Any
+
+
+def get_file_kind(content: Any) -> ProjectFileKind:
+    if is_rulebook_file(content):
+        return ProjectFileKind.RULEBOOK
+    elif is_inventory_file(content):
+        return ProjectFileKind.INVENTORY
+    elif is_playbook_file(content):
+        return ProjectFileKind.PLAYBOOK
+    else:
+        return ProjectFileKind.EXTRA_VARS
+
+
+def scan_file(path: str) -> Optional[ScanResult]:
+    # TODO(cutwater): Should we support both *.yml and *.yaml extensions ?
+    if not path.endswith(".yml"):
+        return None
+
+    try:
+        with open(path) as f:
+            raw_content = f.read()
+    except OSError as exc:
+        logger.warning("Cannot open file %s: %s", os.path.basename(path), exc)
+        return None
 
     try:
+        content = yaml.safe_load(raw_content)
+    except yaml.YAMLError as exc:
+        logger.warning("Invalid YAML file %s: %s", os.path.basename(path), exc)
+        return None
+
+    kind = get_file_kind(content)
+    if kind is None:
+        return None
+
+    return ScanResult(
+        kind=kind,
+        filename=os.path.basename(path),
+        raw_content=raw_content,
+        content=content,
+    )
 
-        await find_rules(project_id, tempdir, db)
-        await find_inventory(project_id, tempdir, db)
-        await find_extra_vars(project_id, tempdir, db)
-        await find_playbook(project_id, tempdir, db)
-        await tar_project(project_id, large_data_id, tempdir, db)
 
-    finally:
-        pass
+def scan_project(project_dir: str) -> Iterator[ScanResult]:
+    for root, dirs, files in os.walk(project_dir):
+        # Skip ignored directories
+        dirs[:] = [x for x in dirs if x not in IGNORED_DIRS]
+        for filename in files:
+            path = os.path.join(root, filename)
+            try:
+                result = scan_file(path)
+            except Exception:
+                logger.exception(
+                    "Unexpected exception when scanning file %s", path
+                )
+                continue
+            if result is not None:
+                yield result
+
+
+def is_rulebook_file(data: Any):
+    if not isinstance(data, list):
+        return False
+    return all("rules" in entry for entry in data)
 
 
-def is_rules_file(filename):
-    if not filename.endswith(".yml"):
+def is_playbook_file(data: Any):
+    if not isinstance(data, list):
         return False
-    try:
-        with open(filename) as f:
-            data = yaml.safe_load(f.read())
-            if not isinstance(data, list):
-                return False
-            for entry in data:
-                if "rules" not in entry:
-                    return False
-    except Exception:
-        logger.exception(filename)
+    for entry in data:
+        if not isinstance(entry, dict):
+            return False
+        if entry.keys() & {"tasks", "roles"}:
+            return True
+    return False
+
+
+def is_inventory_file(data: Any):
```

Comment:
I know that you are preserving the current behavior but I've found a potential bug, because inventories can be also in other formats: https://docs.ansible.com/ansible/latest/plugins/inventory.html#enabling-inventory-plugins
(in fact, `ini` is widely used). 

Yep, the bug would be more in `scan_file` definition. 
I can open a new ticket if you don't want to do it in this PR.

Reply:
yep, this needs to be addressed. I'd prefer to keep it separately.